site/app/views/submission/HomeworkView.php

```diff
@@ -1192,7 +1193,17 @@ public function showRegradeDiscussion(GradedGradeable $graded_gradeable, bool $c
                         break;
                     }
                     $is_staff = $this->core->getQueries()->isStaffPost($post['user_id']);
-                    $name = $this->core->getQueries()->getUserById($post['user_id'])->getDisplayedFirstName();
+                    $is_limited_access_grader = $this->core->getQueries()->isLimitedAccessGraderPost($post['user_id']);
+                    $is_instructor_or_full_access_grader = $this->core->getQueries()->isInstructorOrFullAccessGraderPost($post['user_id']);
```

Comment:
Doing this in a foreach loop is going to cause a *lot* of queries.  Separate queries are very slow, and are the main cause of slowness across Submitty, so we want to minimize the number of queries being made total.  You can click "Show Details" at the bottom of any page in the development environment to see each of the queries being made.  Any repeated queries should be eliminated if at all possible.